exporters/otlp/otlplog/otlploggrpc/client.go

```diff
@@ -71,10 +89,38 @@ func newClient(cfg config) (*client, error) {
 	}
 
 	c.lsc = collogpb.NewLogsServiceClient(c.conn)
-
+	c.initSelfObservability()
 	return c, nil
 }
 
+func (c *client) initSelfObservability() {
+	if !x.SelfObservability.Enabled() {
+		return
+	}
+
+	id := grpcExporterIDCounter.Add(1)
+
+	c.selfObservabilityEnabled = true
+	c.presetAttrs = c.serverAddrAttrs()
+	c.componentName = fmt.Sprintf("%s/%d", otelconv.ComponentTypeOtlpGRPCLogExporter, id-1)
```

Comment:
nit: decrementing here would be more clear

```suggestion
	id := grpcExporterIDCounter.Add(1) - 1

	c.selfObservabilityEnabled = true
	c.presetAttrs = c.serverAddrAttrs()
	c.componentName = fmt.Sprintf("%s/%d", otelconv.ComponentTypeOtlpGRPCLogExporter, id)
```